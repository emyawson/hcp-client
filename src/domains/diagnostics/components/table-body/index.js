export { FixedTableBody, TableBody } from './table-body.component';
