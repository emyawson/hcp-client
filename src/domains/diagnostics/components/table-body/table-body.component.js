import styled from 'styled-components';
import { space } from 'styled-system';

export const TableBody = styled.tbody`
  ${space};
`;

export const FixedTableBody = styled.tbody`
  ${space};
  display: block;
  overflow: auto;
  height: ${props => props.height};
  width: ${props => props.width};
`;
