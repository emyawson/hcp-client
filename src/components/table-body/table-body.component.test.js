import { shallow } from 'enzyme';
import React from 'react';

import { TableBody } from './table-body.component';

describe('table body test suite', () => {
  test('it renders correctly', () => {
    const wrapper = shallow(
      <TableBody>
        <tr>body</tr>
      </TableBody>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
