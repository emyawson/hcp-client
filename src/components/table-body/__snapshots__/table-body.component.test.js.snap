// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`table body test suite it renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.tbody>
    <tr>
      body
    </tr>
  </styled.tbody>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <tr>
        body
      </tr>,
      "className": "sc-bdVaJa eDCunu",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": "body",
      },
      "ref": null,
      "rendered": "body",
      "type": "tr",
    },
    "type": "tbody",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <tr>
          body
        </tr>,
        "className": "sc-bdVaJa eDCunu",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "body",
        },
        "ref": null,
        "rendered": "body",
        "type": "tr",
      },
      "type": "tbody",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
